source/PlayerInfo.cpp

```diff
@@ -989,13 +989,25 @@ Ship *PlayerInfo::Flagship()
 const shared_ptr<Ship> &PlayerInfo::FlagshipPtr()
 {
 	if(!flagship)
+	{
+		bool clearance = false;
+		if(planet)
+		{
+			for(const Mission &mission : missions)
+				if(mission.HasClearance(planet))
+				{
+					clearance = true;
+					break;
+				}
+		}
 		for(const shared_ptr<Ship> &it : ships)
 			if(!it->IsParked() && it->GetSystem() == system && it->CanBeFlagship()
-					&& (!planet || planet->CanLand(*it)))
+				&& (!planet || clearance || planet->CanLand(*it)))
```

Comment:
```suggestion
				&& (!planet || it->GetPlanet() == planet || (clearance && !it->GetPlanet()))
```

Reply:
I don't think so. It should only allow picking a ship not on this planet if that ship isn't landed at all and we have clearance. Though, `clearance` should be changed to `clearance || planet->CanLand(*it)`.

Reply:
Actually, you're right.